core/src/main/resources/hudson/slaves/JNLPLauncher/main.jelly

```diff
@@ -66,10 +117,16 @@ THE SOFTWARE.
                   ${%Run from agent command line:}
                 </p>
                 <!-- TODO conceal secret w/ JS if possible -->
-                <pre>java${it.launcher.vmargs == null ? '' : ' ' + it.launcher.vmargs} -jar <a href="${rootURL}/jnlpJars/slave.jar">slave.jar</a> -jnlpUrl ${h.inferHudsonURL(request)}${it.url}slave-agent.jnlp -secret ${it.jnlpMac}</pre>
+                <textarea class="copiable" readonly="readonly"
+                data-windows="Invoke-WebRequest -Uri $theurl -OutFile \njava${it.launcher.vmargs == null ? '' : ' ' + it.launcher.vmargs} -jar slave.jar -jnlpUrl ${h.inferHudsonURL(request)}${it.url}slave-agent.jnlp -secret ${it.jnlpMac}"
+                data-linux="curl -O ${h.inferHudsonURL(request)}jnlpJars/slave.jar\njava${it.launcher.vmargs == null ? '' : ' ' + it.launcher.vmargs} -jar slave.jar -jnlpUrl ${h.inferHudsonURL(request)}${it.url}slave-agent.jnlp -secret ${it.jnlpMac}">curl -O ${h.inferHudsonURL(request)}jnlpJars/slave.jar
+java${it.launcher.vmargs == null ? '' : ' ' + it.launcher.vmargs} -jar slave.jar -jnlpUrl ${h.inferHudsonURL(request)}${it.url}slave-agent.jnlp -secret ${it.jnlpMac}
+                </textarea>
               </li>
             </j:otherwise>
           </j:choose>
+          <button id="linux"   style="width:10em" onclick="switchText('textarea.copiable[data-linux]','data-linux');  ">Linux</button>
+          <button id="windows" style="width:10em" onclick="switchText('textarea.copiable[data-linux]','data-windows');">Windows</button>
```

Comment:
Actually, I understand your suspicion, but it's not. The right way to do the selection I had in mind would have been something like `textarea.copiable[data-linux&&data-windows]` but I considered it overkill.